### PR TITLE
Add check about filter name for containers

### LINKED ISSUE #21202: Docker ps not valid filter name
**Output of `docker version`:**

```
1.11-dev
```

**Output of `docker info`:**

```
Master branch
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. docker ps -f invalidfilter=test

**Describe the results you received:**
docker ps -f invalidfilter=work not valid such filter

**Describe the results you expected:**
docker ps should valid filter like images, network, etc.

**Additional information you deem important (e.g. issue happens only occasionally):**

### PR DESCRIPTION
**Pull request by HackToday**
Fixes: #21202
We add valid check about filters like network, images,
volumes did.

Signed-off-by: Kai Qiang Wu(Kennan) wkqwu@cn.ibm.com

### CODE REVIEW

File: daemon/list.go [hunk 1]
```diff
@@ -20,6 +20,18 @@ var acceptedVolumeFilterTags = map[string]bool{
...
+var acceptedPsFilterTags = map[string]bool{
+	"exited":    true,
+	"label":     true,
+	"status":    true,
+	"ancestor":  true,
+	"name":      true,
+	"id":        true,
+	"before":    true,
+	"since":     true,
+	"isolation": true,
```
Comment (thaJeztah) [inline daemon/list.go]: Can you out these in alphabetical order?

**Comment by thaJeztah** [comment]
Looks like you forgot to add `volume` to the list of filters;

```
04:50:57 FAIL: docker_cli_ps_test.go:761: DockerSuite.TestPsShowMounts
04:50:57 
04:50:57 docker_cli_ps_test.go:786:
04:50:57     // filter by volume name
04:50:57     out, _ = dockerCmd(c, "ps", "--format", "{{.Names}} {{.Mounts}}", "--filter", "volume=ps-volume-test")
04:50:57 /go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils.go:42:
04:50:57     c.Assert(err, check.IsNil, check.Commentf(quote+"%v"+quote+" failed with errors: %s, %v", strings.Join(args, " "), out, err))
04:50:57 ... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc823773ae0), Stderr:[]uint8(nil)} ("exit status 1")
04:50:57 ... "ps --format {{.Names}} {{.Mounts}} --filter volume=ps-volume-test" failed with errors: Error response from daemon: Invalid filter 'volume'
```

**Comment by HackToday** [comment]
ping @thaJeztah PTAL the new change

File: daemon/list.go [hunk 2]
```diff
@@ -129,6 +142,10 @@ func (daemon *Daemon) foldFilter(config *types.ContainerListOptions) (*listConte
 	psFilters := config.Filter
 
 	var filtExited []int
+	if err := psFilters.Validate(acceptedPsFilterTags); err != nil {
```
Comment (thaJeztah) [inline daemon/list.go]: nit, but should this be above `var filtExited ..` ?

**Comment by thaJeztah** [comment]
one minor nit, but LGTM

**Comment by vdemeester** [comment]
LGTM 🐯 (after the nit taken care of)

**Comment by runcom** [comment]
are those filters documented anywhere? just wondering otherwise code LGTM, @thaJeztah I'll move to doc review, ping @calavera @cpuguy83 also

**Comment by thaJeztah** [comment]
@runcom they are, since earlier today :wink:  https://github.com/docker/docker/pull/21209/files

**Comment by runcom** [comment]
Cool :) thx for checking!

**Comment by HackToday** [comment]
ping @thaJeztah and @vdemeester  PTAL the new change.

**Comment by HackToday** [comment]
@runcom  seems we not need to add  status/3-docs-review

**Comment by thaJeztah** [comment]
Hm flaky test? https://jenkins.dockerproject.org/job/Docker-PRs-WoW-TP4/2941/console

```
05:40:38 PASS: docker_cli_run_test.go:536: DockerSuite.TestRunApplyVolumesFromBeforeVolumes	42.889s
05:41:34 
05:41:34 ----------------------------------------------------------------------
05:41:34 FAIL: docker_cli_run_test.go:4223: DockerSuite.TestRunAttachFailedNoLeak
05:41:34 
05:41:34 docker_cli_run_test.go:4230:
05:41:34     c.Assert(err, checker.NotNil)
05:41:34 ... value = nil
05:41:34 
05:41:56 
05:41:56 ----------------------------------------------------------------------
```

**Comment by thaJeztah** [comment]
LGTM

docs is not needed indeed

**Comment by vdemeester** [comment]
LGTM 🐮

**Pull request merged**